mutual_neighborhood_graph.h

```diff
@@ -334,7 +334,7 @@ void cluster_remaining(
         }
     }
     for (int c = 0; c < n_chosen_clusters; ++c) {
-        for (int i = 0; i < modal_sets[c].size() - 1; ++i) {
+        for (int i = 0; (unsigned) i < modal_sets[c].size() - 1; ++i) {
```

Comment:
Does it make more sense to just do something like
for (unsigned int i = 0; ... ) ?

Reply:
That's an interesting thought. I personally don't mind either way but I chose to do it this way to comply with google's style guidelines on using unsigned variables. If we're going to change the type, perhaps we should consider `size_t` instead of `unsigned int` which would have better portability guarantees.

What do you think?

Reply:
size_t sounds good

Reply:
Cool, updated.